### PR TITLE
1522 Information Marker

### LINKED ISSUE #1522: UX for Information Markers
The `Game Controller` can now add and edit info markers.  Background [here](https://docs.google.com/presentation/d/1AOGfEQuSYsRyvAYUe4_x3E_bvu15KJ8KpzqDyDAMh-Q/edit#slide=id.g113e814bd03_0_0)

The form will probably be similar to the Perception Form:
image

Here is the flow:
- If logged in as Game Controller, an `Add information marker` icon is shown in the map-control-bar (see below)
- When this button is clicked, a new marker is placed at the centre of the current view
- Once added, the marker is selected, and the `Edit marker form` is shown.
- This form allows the Game Controller to change these properties:
    - label
    - description (text-area)
    - icon (from drop-down list. These aren't platform-types, they are a separate list that are maintained in a new Game Admin tab)
    - visible to (list of check-boxes, one per force)
    - radius (integer, default value zero)
    - Color (this may be the color of a force, or any other color - since the Game Control may specify that icons of particular color relate to same subjects, such as `marine mammals`, `civilian interactions`, `weather features`)
- Once edits are complete, the Game Controller clicks on `Save`

image

### PR DESCRIPTION
**Pull request by lebaphi**
Fixes #1522

### CODE REVIEW

File: packages/components/src/local/adjudicate-turn-form/__snapshots__/index.spec.tsx.snap [hunk 1]
```diff
@@ -121,7 +121,7 @@ exports[`AdjudicateTurnForm renders correctly 1`] = `
         className="container main-container align-right"
       >
         <section
-          className="input-container"
+          className="input-container input-container"
```
Comment (IanMayo) [inline packages/components/src/local/adjudicate-turn-form/__snapshots__/index.spec.tsx.snap]: @lebaphi - do you know why the class name is being provided twice?

Comment (lebaphi) [reply packages/components/src/local/adjudicate-turn-form/__snapshots__/index.spec.tsx.snap]: @IanMayo I'm not sure, it's auto generated by jest 🤔

Comment (IanMayo) [reply packages/components/src/local/adjudicate-turn-form/__snapshots__/index.spec.tsx.snap]: Sorry, of course it is. Let's not worry about it.

**Comment by IanMayo** [comment]
Hello @lebaphi -  I'm just finishing a branch/PR that renders the "radius" polygons for the Information Markers:
https://github.com/serge-web/serge/pull/1559

image


Once it's done, you should merge it into this branch, so you can check the "radius" edit is working properly

**Comment by lebaphi** [comment]
okay @IanMayo , Im done the form marker side, Im working on the Marker Icons tab in Admin side.

**Comment by IanMayo** [comment]
Ok.  can we do the admin page in a separate PR please?

Note: @ShendMikullovci & I have agreed the UX.  The `label` and `icon` fields will be editable from the form:
https://github.com/serge-web/serge/issues/1522#issuecomment-1127461065

**Comment by lebaphi** [comment]
Hi @IanMayo , it works

https://user-images.githubusercontent.com/26705621/168715628-950246f6-ce39-4ed4-bd67-947026a8ebf8.mov

**Comment by IanMayo** [comment]
That video is great, thanks @lebaphi .

I've realised we've missed something. We don't have the color editor in there. Have added thoughts here: https://github.com/serge-web/serge/issues/1522#issuecomment-1128630359

**Comment by IanMayo** [comment]
Thanks @lebaphi  - note: it is also missing the ability to change the `icon`.  The list of icons should come from `data.annotationIcons` - and show both the icon name and the image in the drop-down.

**Comment by lebaphi** [comment]
I think we need the BE ready first then we can go back to update the list of icon 😉

**Comment by IanMayo** [comment]
Hello @lebaphi  - I've just looked at the code.  The component is receiving the list of icons (`formdata.populate.icons`).  So, I don't think we need any back-end changes.  I think this PR can be modified to include the drop-down list of icons, and the popup color editor.

**Comment by IanMayo** [comment]
Ok.  The marker form now has color & icon.

I haven't been able to connect to the popup color editor we use in the `settings-forces` admin page.

So, could you please:
- connect up color editing
- get the layout closer to what @ShendMikullovci specified

**Comment by IanMayo** [comment]
Oops @lebaphi , code changes in `componentSelector` mean that the UI crashes when we click on an asset that wants to open the `adjudicate turn form`.  The error is _something_ to do with the `useStyles` in `componentSelector.tsx`.  It's called from the `<RCB>` component in adjudicate-turn-form.

Could you investigate this please?  The fix looks really tricky. If you can't solve it, we'll have to find another way of specifying the color of check-boxes.

screencast

**Comment by lebaphi** [comment]
no worries @IanMayo , I will take care it for you 😉

**Pull request merged**